src/lib.rs

```diff
@@ -1,4 +1,27 @@
+//! Switch-HAL is a no_std embedded Rust library for working with buttons, switches, LEDs, and transistors. Basically, anything that acts like a switch, whether an input or output.
+//! 
+//! It is both a driver that uses the embedded-hal::digital traits and is an abstraction in it's own right. It provides a simple, zero-cost, abstraction to clarify the intent of your application code.
+//! 
+//! Why Switch-HAL? Why not just use raw GPIO?
+//! Did you mean to drive that line high?
+//! Or did you mean to turn that LED off?
+//! Wait a second... is that LED active high?
+//! Where's the schematic?
+//! Okay... cathode is wired to the input line... that means it's active low.
+//! 
+//! Now repeat this every place in your code where you need to turn that LED on or off.
+//! What happens when the hardware changes?
+//! Using the raw GPIO to set pins high and low will have you making changes all over your code base.
+//! 
+//! Wouldn't it be nicer if you only had to think about that once, when you initialize your application,
+//! and from then on out, simply called led.on() or led.off().
+//! Having an abstraction at the proper level reduces cognitive load.
+//! Specifying whether a simple peripheral is active high or low in a single place in your application reduces the maintenance burden.
+//! 
+//! "Async" feature enables async support for input switches.
```

Comment:
I find copying the elevator pitch from the readme into the module docs unnecessary and don’t really want to maintain the duplication.